ttls/client.py

```diff
@@ -164,9 +167,39 @@ async def interview(self, force: bool | None = False) -> None:
                 self.default_mode = mode.get("mode")
 
     def _get_session(self):
-        return self._session or ClientSession()
+        if not self._session:
+            self._session = ClientSession() 
+        return self._session
+
+    async def _info(self) -> Any:
+        _LOGGER.debug("INFO")
+        try:
+            async with self._get_session().get(
+                f"http://{self.host}/xled/info",
+                timeout=self._timeout,
+                raise_for_status=True,
+            ) as r:
+                _LOGGER.debug("INFO response %d", r.status)
+                return await r.json()
+        except ClientResponseError as e:
+                raise e
+
+    async def get_api_version(self) -> int:
+        if self._api_version is not None:
+            return self._api_version
+        return await self.detect_api_version()
+
+    async def detect_api_version(self) -> int:
+        try:
+            info = await self._info()
+            return int(info['api']['min_version'].split('.')[0])
+        except ClientResponseError as e:
+            _LOGGER.debug(e)
+            return 1
 
     async def _post(self, endpoint: str, **kwargs) -> Any:
+        if self._api_version is None:
+            self._api_version = await self.get_api_version()
```

Comment:
Perhaps define `api_version` as a property? If not set when requested, detect it and set it.

Reply:
Right, async properties are hard.